ome/datasource_device.go

```diff
@@ -0,0 +1,196 @@
+/*
```

Comment:
I will provide a comprehensive review of the provided pull request diff and suggest improvements.